moto/ec2/models.py

```diff
@@ -3831,19 +3831,173 @@ def describe_vpc_endpoints(self, vpc_end_point_ids, filters=None):
 
         return generic_filter(filters, vpc_end_points)
 
-    def get_vpc_end_point_services(self):
-        vpc_end_point_services = self.vpc_end_points.values()
+    @staticmethod
+    def _collect_default_endpoint_services(region):
+        """Return list of default services using list of backends."""
+        zones = [
+            zone.name
+            for zones in RegionsAndZonesBackend.zones.values()
+            for zone in zones
+            if zone.name.startswith(region)
+        ]
+
+        # Rather than import this function in each of the backends, this
+        # function will be passed as an argument.
+        def vpce_random_number():
+            """Create a 17 digit number of lowercase hex."""
+            return "".join([random.choice(string.hexdigits.lower()) for i in range(17)])
+
+        from moto import backends  # pylint: disable=import-outside-toplevel
+
+        default_services = []
+        for _backends in backends.unique_backends():
+            for region_name, backend in _backends.items():
```

Comment:
Potential performance improvement:
Instead of iterating over all regions, we could just do with two ifs.
```
if region_name in _backends:
    ....
if "global" in _backends:
    ...
```

Reply:
Doh!  Yes, I can use the region as a key.  I struggled with that function and get a bit lost.